compiler/rustc_expand/src/mbe/transcribe.rs

```diff
@@ -694,7 +690,20 @@ fn transcribe_metavar_expr<'a>(
                 let string = match element {
                     MetaVarExprConcatElem::Ident(elem) => elem.to_string(),
                     MetaVarExprConcatElem::Literal(elem) => elem.as_str().into(),
-                    MetaVarExprConcatElem::Var(elem) => extract_ident(dcx, *elem, interp)?,
+                    MetaVarExprConcatElem::Var(ident) => {
+                        match matched_from_ident(dcx, *ident, interp)? {
+                            NamedMatch::MatchedSeq(named_matches) => {
+                                let curr_idx = repeats.last().unwrap().0;
+                                match &named_matches[curr_idx] {
+                                    MatchedSeq(_) => unimplemented!(),
```

Comment:
Currently only supports single repetitions. Nested repetitions are more complicated and I prefer to deal with them in a following PR.

Reply:
Please leave a fixme comment in code.

Reply:
Done